File: Easy/217-Contains-Duplicate.py
# Task: Given an integer array nums, 
# return true if any value appears at least twice in the array, 
# and return false if every element is distinct.

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = []
        storing_data = sorted(nums)
        for number in storing_data:
            if number in hashset:
                return True
            hashset.append(number)
        return False

File: Easy/217-Contains-Duplicate.rs
// Task: Given an integer array nums,
// return true if any value appears at least twice in the array,
// and return false if every element is distinct.

use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut our_hashset = HashSet::new();
        for &number in nums.iter() {
            if our_hashset.contains(&number) {
                return true;
            }

            our_hashset.insert(number);
        }

        return false;
    }
}
